packages/react-dom/src/__tests__/ReactDOMEventListener-test.js

```diff
@@ -33,54 +33,53 @@ describe('ReactDOMEventListener', () => {
   });
 
   it('should dispatch events from outside React tree', () => {
+    var mock = jest.fn();
     var otherNode = document.createElement('h1');
-    var component = ReactDOM.render(<div />, document.createElement('div'));
-    expect(handleTopLevel.mock.calls.length).toBe(0);
-    ReactDOMEventListener.dispatchEvent('topMouseOut', {
-      type: 'mouseout',
-      fromElement: otherNode,
-      target: otherNode,
-      srcElement: otherNode,
-      toElement: ReactDOM.findDOMNode(component),
-      relatedTarget: ReactDOM.findDOMNode(component),
-      view: window,
-      path: [otherNode, otherNode],
-    });
-    expect(handleTopLevel.mock.calls.length).toBe(1);
+    var component = ReactDOM.render(
+      <div onMouseOver={mock} />,
+      document.createElement('div'),
+    );
+
+    ReactDOM.findDOMNode(component).appendChild(otherNode);
+    ReactTestUtils.SimulateNative.mouseOver(otherNode);
```

Comment:
For these purposes, I'd like to treat `SimulateNative` (and all the ReactTestUtils really as an internal API. The reason for that is that they also reach into internals indirectly. They actually bypass some of the implementation details that would make this example not work with normal React.

In normal React this wouldn't work because the container div is not in the document and therefore mouse moves would never be dispatched on top of it. React can't pick it up because it uses event listeners on the document to do it.

Instead, we should insert the container div into the document, dispatch a native event and then remove the div from the document. Like this:
https://github.com/facebook/react/blob/master/packages/react-dom/src/__tests__/ReactDOMInput-test.js#L55-L59

Reply:
Hey! I did the requested changes, could you take a look at it?
I'm still not sure if I'm the first test is on spot because before it tested two unrelated components and now they have a parent-child relationship. 

Also while we are at this file, I'm not sure how to rewrite `should not get confused by disappearing elements`.. I think I understand the thinking behind it but I'm not sure if it's possible to accomplish the same result without mocking `handleTopLevel`. I'd have to unmount a component in the middle of the event chain, which seems impossible.

Thanks!